src/components/textinput/createTextInput.ts

```diff
@@ -10,7 +10,11 @@ export interface TypedTargetEvent<T extends EventTarget> extends Event {
 
 export type TextInputState = WidgetState & FormFieldMixinState<string>;
 
-export type TextInputOptions = WidgetOptions<TextInputState, WidgetProperties> & FormFieldMixinOptions<string, TextInputState>;
+export interface TextInputProperties extends WidgetProperties {
+	type?: string;
```

Comment:
I don't personally think it should. I'm open to not enforcing that, if you think it should be accessed via `state`